src/environment.py

```diff
@@ -141,10 +172,77 @@ def step(self, action: int) -> Tuple[int, np.ndarray, bool]:
         print("reward", reward)
         next_state = self.get_state()
 
-        # TODO: Update enemy positions
-        # self.enemies.step()?
+        for enemy in self.enemies:
+            enemy.move()
+
+            # Check if the enemy is outside the screen and remove it only if it's moving away from the screen
+            # This ensures that self.enemies is not getting infinitely large
+            if self.is_moving_outside_the_screen(enemy):
+                self.enemies.remove(enemy)
 
         if self.render_on:
             self.render()
 
         return reward, next_state, done
+
+    def is_moving_outside_the_screen(self, enemy: Enemy) -> bool:
```

Comment:
I feel like this function is very similar to the `is_valid_location()` function, as both check if an agent is going outside the screen. Maybe you could update that function to be more generic and reuse it for the enemies as well? I do like this name, it is more descriptive than `is_valid_location()`.